librocksdb-sys/build.rs

```diff
@@ -357,9 +355,40 @@ fn update_submodules() {
     }
 }
 
+fn update_submodules() {
+    let program = "git";
+    let dir = "../";
+    let args = ["submodule", "update", "--init"];
+    run_command(program, dir, &args);
+}
+
+fn cherry_pick_commits(commits: &Vec<&str>) {
+    for commit in commits {
+        let program = "git";
+        let dir = "rocksdb/";
+        let args = ["cherry-pick", &commit];
+        run_command(program, dir, &args);
+    }
+}
+
+fn apply_patches(patches: &Vec<&str>) {
```

Comment:
```suggestion
fn apply_patches(patches: &[&str]) {
```

Reply:
I don't know to to get a slice from a split function, it won't work just like this

Reply:
`split` returns an iterator.  When you use `collect`, you need to sometimes provide type hints if the compiler is not able to infer the actual types.  Something like the following works.

```
fn foo(arg: &[&str]) {
    println!("{:?}", arg);
}


fn main() {
    let arg = "hello world";
    foo(&arg.split(" ").collect::<Vec<&str>>());
}
```

Reply:
In general, slices are preferred over `Vec<T>` because slices are more generic i.e. more things can be converted into them so it makes your function able more inclusive.